modules/persistent-storage-csi-vsphere-change-max-snapshot.adoc

```diff
@@ -0,0 +1,130 @@
+// Module included in the following assemblies:
+//
+// * storage/persistent_storage/persistent-storage-vsphere.adoc
+// * storage/persistent_storage/persistent-storage-csi-snapshots.adoc
+
+:_mod-docs-content-type: PROCEDURE
+[id="vsphere-change-max-snapshot_{context}"]
+= Changing the maximum number of snapshots for vSphere
+
+The default maximum number of snapshots per volume in vSphere Container Storage Interface (CSI) is 3. You can change the maximum number up to 32 per volume. 
+
+However, be aware that increasing the snapshot maximum involves a performance trade off, so for better performance use only 2 to 3 snapshots per volume.
+
+For more VMWare snapshot performance recommendations, see *_Additional Resources_*.
+
+.Prerequisites
+
+* Access to the cluster with administrator rights.
+
+.Procedure
+
+. Check the current config map by the running the following command:
++
+[source, terminal]
+----
+$ oc -n openshift-cluster-csi-drivers get cm/vsphere-csi-config -o yaml
+----
++
+.Example output
++
+[source, terminal]
+----
+apiVersion: v1
+data:
+  cloud.conf: |+
+    # Labels with topology values are added dynamically via operator
+    [Global]
+    cluster-id = rbednar-vsphere-01-cwv8p
+
+    [VirtualCenter "vcenter.devqe.ibmc.devcluster.openshift.com"]
+    insecure-flag           = true
+    datacenters             = DEVQEdatacenter
+    migration-datastore-url = ds:///vmfs/volumes/vsan:527320283a8c3163-2faa6dc5949a3a28/
+
+kind: ConfigMap
+metadata:
+  creationTimestamp: "2024-03-06T09:46:40Z"
+  name: vsphere-csi-config
+  namespace: openshift-cluster-csi-drivers
+  resourceVersion: "126687"
+----
++
+In this example, the global maximum number of snapshots is not configured, so the default value of 3 is applied.
+
+. Change the snapshot limit by running the following command:
++
+* Set *global* snapshot limit:
++
+[source, terminal]
+----
+$ oc patch clustercsidriver/csi.vsphere.vmware.com --type=merge -p '{"spec":{"driverConfig":{"vSphere":{"globalMaxSnapshotsPerBlockVolume": 10}}}}'
+clustercsidriver.operator.openshift.io/csi.vsphere.vmware.com patched
+----
++
+In this example, the global limit is being changed to 10 (`globalMaxSnapshotsPerBlockVolume` set to 10).
+
+* Set *Virtual Volume* snapshot limit:
++
+[source, terminal]
+----
+$ oc patch clustercsidriver/csi.vsphere.vmware.com --type=merge -p '{"spec":{"driverConfig":{"vSphere":{"granularMaxSnapshotsPerBlockVolumeInVVOL": 5}}}}'
+clustercsidriver.operator.openshift.io/csi.vsphere.vmware.com patched
+----
++
+In this example, the Virtual Volume limit is being changed to 5 (`granularMaxSnapshotsPerBlockVolumeInVVOL` set to 5).
```

Comment:
no really clear IMO. You can look at the vsphere CSI doc for inspiration
https://docs.vmware.com/en/VMware-vSphere-Container-Storage-Plug-in/3.0/vmware-vsphere-csp-getting-started/GUID-E0B41C69-7EEB-450F-A73D-5FD2FF39E891.html#GUID-7BA0CDAE-E031-470E-A685-60C82DAE36D2__GUID-D9A97A90-2777-46EA-94EB-F04A27FBB76D

Reply:
Added some info.